warp/localized.py

```
@@ -0,0 +1,1469 @@
+    print(f"VMM: Reserving {aligned_total_size} bytes of virtual address space")
+
+    # Reserve virtual address space ONCE for the entire allocation
+    err, base_dptr = cuda_driver.cuMemAddressReserve(
+        aligned_total_size,
+        granularity,
+        0,  # addr (0 = let CUDA choose)
+        0,  # flags
+    )
+
+    if err != cuda_driver.CUresult.CUDA_SUCCESS:
+        raise RuntimeError(f"cuMemAddressReserve failed with error {err}")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)